File: Zad0/zad0.cpp
#include <stdio.h>
#include <json/value.h>
#include <fstream>

using namespace std;

vector <char> alphabet;
vector <string> states;
string initial;
unordered_set <string> accepting;
map <pair <string, char>, string> transitions;

int main(){
    //WCZYTAJ JSON!
    string current = initial;
    char c = getchar();
    while(c!= EOF){
        if(c=='\n'){
            if(accepting.find(current)!=accepting.end())
                printf("yes\n");
            else
                printf("no\n");
            current = initial;
            continue;
        }
        current = transitions[make_pair(current, c)];
        c = getchar();
    }
}

File: Zad0/zad0.py
import json
import sys

path = input()
with open(path, 'r') as f:
    automaton = json.load(f)

accepting = automaton["accepting"]

dict = {}
for t in automaton["transitions"]:
   dict[(t["letter"], t["from"])] = t["to"]

state = automaton["initial"]

c = sys.stdin.read(1)

while c:
    if c != '\n':
        state = dict[(c, state)]
    else:
        if state in accepting:
            print("yes")
        else:
            print("no")
        state = automaton["initial"]
    c = sys.stdin.read(1)
